lib/pbench/test/unit/server/test_api_key.py

```diff
@@ -90,4 +92,162 @@ def test_successful_api_key_generation(self, query_get_as, pbench_drb_token):
         assert audit[1].user_id == "3"
         assert audit[1].user_name == "drb"
         assert audit[1].reason is None
-        assert audit[1].attributes["key"] == response.json["api_key"]
+        assert audit[1].attributes["key_id"] == response.json["id"]
+
+
+class TestAPIKeyGet:
+    @pytest.fixture()
+    def query_get_as(self, client, server_config):
+        """
+        Helper fixture to perform the API query and validate an expected
+        return status.
+
+        Args:
+            client: Flask test API client fixture
+            server_config: Pbench config fixture
+        """
+
+        def query_api_get(user_token, expected_status: HTTPStatus) -> requests.Response:
+            headers = {"authorization": f"bearer {user_token}"}
+            response = client.get(
+                f"{server_config.rest_uri}/key",
+                headers=headers,
+            )
+            assert response.status_code == expected_status
+            return response
+
+        return query_api_get
+
+    def test_successful_api_key_get(
+        self,
+        query_get_as,
+        pbench_drb_token,
+        pbench_drb_api_key,
+        pbench_drb_secondary_api_key,
+    ):
+
+        response = query_get_as(pbench_drb_token, HTTPStatus.OK)
+
+        assert response.json["api_key"]
+        assert len(response.json["api_key"]) == 2
+        assert response.json["api_key"]["1"]["name"] == "drb_key"
+        assert response.json["api_key"]["1"]["key"] == pbench_drb_api_key.api_key
+        assert response.json["api_key"]["2"]["name"] == "secondary_key"
+        assert (
+            response.json["api_key"]["2"]["key"] == pbench_drb_secondary_api_key.api_key
+        )
+
+    def test_unauthorized_get(self, query_get_as, pbench_drb_token_invalid):
+        response = query_get_as(pbench_drb_token_invalid, HTTPStatus.UNAUTHORIZED)
+        assert response.json == {
+            "message": "User provided access_token is invalid or expired"
+        }
+
+
+class TestAPIKeyDelete:
+    @pytest.fixture()
+    def query_as(self, client, server_config):
```

Comment:
For consistency, this might be `query_delete_as`. Sigh: these are all over the place, and I wonder how much they could be generalized and consolidated. 😦 Unfortunately, "configuring" fixtures is awkward... although it's really almost silly that these were originally packaged as fixtures instead of just defining the inner method and calling it directly.